modules/command-engine/core/src/main/scala/surge/internal/persistence/KTablePersistenceSupport.scala

```diff
@@ -46,7 +50,16 @@ trait KTablePersistenceSupport[Agg, Event] {
       serializedState: KafkaProducerActor.MessageToPublish,
       startTime: Instant)
       extends Internal
-  private case class EventPublishTimedOut(reason: Throwable, startTime: Instant) extends Internal
+  private case class EventPublishTimedOut(
+      trackingId: UUID,
```

Comment:
I think this needs to track a `newState` in this object since that's eventually used in the case of persistence success, the `state` passed into `handleInternal` is the old state (in case of failure)

Reply:
done